src/expr/src/relation/func.rs

```diff
@@ -978,6 +978,59 @@ fn generate_series_ts(
     Ok(tsri.map(move |i| (Row::pack_slice(&[conv(i)]), 1)))
 }
 
+fn generate_subscripts_array(
+    a: Datum,
+    dim: i32,
+) -> Result<Box<dyn Iterator<Item = (Row, Diff)>>, EvalError> {
+    if dim <= 0 {
+        return Ok(Box::new(iter::empty()));
+    }
+
+    match a.unwrap_array().dims().into_iter().nth(
+        (dim - 1)
+            .try_into()
+            .expect("array dimensions must be a usize"),
+    ) {
+        Some(requested_dim) => generate_series::<i32>(
+            1,
```

Comment:
The lower argument here should be `requested_dim.lower_bound` (even though I don't think it's possible at the moment to construct an array with a lower bound of something besides 1).

Reply:
can you explain in which scenario we might want to generate a series that does not start with 1?

Reply:
In the situation that the array's lower bound is not 1. E.g. in Postgres:

```
benesch=# select generate_subscripts('[2:5]={1,2,3,4}'::int[], 1);
 generate_subscripts 
---------------------
                   2
                   3
                   4
                   5
(4 rows)
```

There's some sparse documentation about this in the "Arrays" chapter of the PostgreSQL manual: https://www.postgresql.org/docs/current/arrays.html

Reply:
we don't yet support the syntax. I did change the code to honor the lower bound in case this support will be present in the future. I also added a test to verify that the query fails, it should remind us to fix it, if the limitation is ever removed. Well spotted Nikhil!